vumi/persist/riak_manager.py

```diff
@@ -97,69 +67,22 @@ def get_index_page(self, index_name, start_value, end_value=None,
         return VumiIndexPage(result)
 
 
-class VumiRiakObject(object):
-    def __init__(self, riak_obj):
-        self._riak_obj = riak_obj
-
-    @property
-    def key(self):
-        return self._riak_obj.key
-
-    def get_key(self):
-        return self.key
-
-    def get_content_type(self):
-        return self._riak_obj.content_type
-
-    def set_content_type(self, content_type):
-        self._riak_obj.content_type = content_type
-
-    def get_data(self):
-        return self._riak_obj.data
-
-    def set_data(self, data):
-        self._riak_obj.data = data
-
-    def set_encoded_data(self, encoded_data):
-        self._riak_obj.encoded_data = encoded_data
-
-    def set_data_field(self, key, value):
-        self._riak_obj.data[key] = value
-
-    def delete_data_field(self, key):
-        del self._riak_obj.data[key]
-
-    def get_indexes(self):
-        return self._riak_obj.indexes
-
-    def set_indexes(self, indexes):
-        self._riak_obj.indexes = indexes
-
-    def add_index(self, index_name, index_value):
-        self._riak_obj.add_index(index_name, index_value)
-
-    def remove_index(self, index_name, index_value=None):
-        self._riak_obj.remove_index(index_name, index_value)
-
-    def get_user_metadata(self):
-        return self._riak_obj.usermeta
-
-    def set_user_metadata(self, usermeta):
-        self._riak_obj.usermeta = usermeta
+class VumiRiakObject(VumiRiakObjectBase):
+    """
+    Wrapper around a RiakObject to manage network access better.
+    """
 
     def get_bucket(self):
         return VumiRiakBucket(self._riak_obj.bucket)
 
     # Methods that touch the network.
 
-    def store(self):
-        return type(self)(self._riak_obj.store())
-
-    def reload(self):
-        return type(self)(self._riak_obj.reload())
-
-    def delete(self):
-        return type(self)(self._riak_obj.delete())
+    def _call_and_wrap(self, func):
+        """
+        Call a function that touches the network and wrap the result in this
+        class.
+        """
+        return type(self)(func())
```

Comment:
My python knowledge is being stretched here... but why can't this (`return type(self)(func())`) be in the base class?